assets/src/dashboard/app/reducer/stories.js

```diff
@@ -151,7 +151,9 @@ function storyReducer(state, action) {
     case ACTION_TYPES.FETCH_STORIES_SUCCESS: {
       const fetchedStoriesById = [];
       const reshapedStories = action.payload.stories.reduce((acc, current) => {
-        if (!current) {
+        // Hardening in case data returned by the server is malformed.
+        // For example, story_data could be missing.
+        if (!current || !current?.story_data?.length) {
```

Comment:
This actually feels like something that should be caught before getting to the reducer. What about this in the `useStoryApi` - around line 104. 

```
 const cleanStories = response.body.filter(
          (story) => story?.story_data?.length > 0
        );

        if (cleanStories.length <= 0) {
          dispatch({
            type: STORY_ACTION_TYPES.FETCH_STORIES_FAILURE,
            payload: {
              message: {
                body: _n('Story Data Missing', 'web-stories'),
                title: ERRORS.LOAD_STORIES.TITLE,
              },
            },
          });
        } else {
          dispatch({
            type: STORY_ACTION_TYPES.FETCH_STORIES_SUCCESS,
            payload: {
              editStoryURL,
              stories: cleanStories,
              totalPages,
              totalStoriesByStatus: {
                ...totalStoriesByStatus,
                [STORY_STATUS.PUBLISHED_AND_FUTURE]:
                  totalStoriesByStatus[STORY_STATUS.PUBLISH] +
                  totalStoriesByStatus[STORY_STATUS.FUTURE],
              },
              page,
            },
          });
        }
      } catch (err) {
...
```

Reply:
Ah yes that looks much cleaner 👍